### ISSUE TITLE
NPE Exception in pauseDownload reported by Google Play Console

### ISSUE THREAD

**Issue by julianharty** [kiwix/kiwix-android#712]
Google Play Console provides lots of information about the Kiwix-Android app available in the Google Play store (rather than ones users install from elsewhere).

The app has an overall crash rate of 1.53% currently on user's devices. However the crashes aren't evenly distributed so some users and some devices experience far higher crash rates than others.

Many of the most frequent crashes are related to the Download process. And of these the `pauseDownload` is the most common. I will focus on this feature here. We can raise issues for `stopDownload` and other major causes so we can work on these orthogonally and track our progress.

I've raised this issue so more of us know about what's happening for end-users in the hope that we'll be able to identify and fix the cause(s).

## Google Play Crashes ranked by most crashes

<img width="726" alt="screen shot 2018-04-13 at 14 26 19" src="https://user-images.githubusercontent.com/785891/38737302-b1d22bc4-3f26-11e8-8ffd-0b1a8ab4deb7.png">

The essence of the stack trace is:
```
Caused by: java.lang.NullPointerException: 
  at org.kiwix.kiwixmobile.downloader.DownloadService.pauseDownload (DownloadService.java:196)
  at org.kiwix.kiwixmobile.downloader.DownloadService.toggleDownload (DownloadService.java:190)
  at org.kiwix.kiwixmobile.downloader.DownloadService.onStartCommand (DownloadService.java:105)
```
## Screenshots of the crash on various Android versions
For whatever reasons I don't fathom, Google Play has 3 groupings of the same crash. Here are the 3 screenshots of these.
The most frequent is reported on older Android versions 6.0 and earlier
<img width="732" alt="screen shot 2018-04-13 at 14 10 56" src="https://user-images.githubusercontent.com/785891/38736682-beebd46a-3f24-11e8-86a1-3d6144b68119.png">

The next grouping is for 7.0 and 7.1 devices
<img width="725" alt="screen shot 2018-04-13 at 14 14 43" src="https://user-images.githubusercontent.com/785891/38736786-0c9eb506-3f25-11e8-8638-21c4e68f1a15.png">

Finally followed by those on Android 8.0, 7.0 and 7.1 (presumably these don't overlap with the previous grouping) 

<img width="730" alt="screen shot 2018-04-13 at 14 15 28" src="https://user-images.githubusercontent.com/785891/38736860-53dacfe0-3f25-11e8-869f-15e3212338c9.png">

### Environment
Production: reported by Google Play Console.
  * version of the software you use : Release 2.3 and 2.2 (although the 
  * device / operating system : Various end-user devices.

**Comment by julianharty** [comment]
## Testing notes
The automated tests don't test either pauses or stopping and restarting downloads. These tests would be welcome, especially once we get the current downloadTest to run reliably. However we probably don't want to have long running tests in our CI process so we may want/need to create a separate test group that is skipped by Travis-CI / Testdroid.

The problems are probably better to test interactively initially. Perhaps adding some logging in the app code in `DownloadService.java` would help identify what's triggering the NPE? I'll do some investigatory testing, you're welcome to contribute too.

**Comment by mhutti1** [comment]
I think this is in general related to the notification not interacting with underlying service correctly in certain cases where some fragments are not loaded.

**Comment by macgills** [comment]
Fixed in #1170
